File: py/1496.py
class Solution:
    def isPathCrossing(self, path: str) -> bool:
        been = []

        location = [0, 0]

        been.append(location)

        for n in path:
            temp = [n for n in location]

            match n:
                case 'N':
                    temp[1] += 1
                case 'S':
                    temp[1] -= 1
                case 'E':
                    temp[0] += 1
                case 'W':
                    temp[0] -= 1

            if temp in been:
                return True

            been.append(temp)
            location = temp

        return False

File: go/1496.go
type IntPair struct {
    x int
    y int
}

func isPathCrossing(path string) bool {
    visited := make(map[IntPair]bool)

    location := IntPair{0,0}

    visited[location] = true

    for _,direction := range path {
         if direction == 'N' {
             location.y--
         } else if direction == 'S' {
             location.y++
         } else if direction == 'W' {
             location.x--
         } else if direction == 'E' {
             location.x++
         }

         _,ok := visited[location]
         if ok {
           return true
         } else {
             visited[location] = true
         }
    }
    return false
}
